.github/workflows/e2e-matrix-nap.yaml

```diff
@@ -0,0 +1,61 @@
+name: E2EMatrixNAP
+on:
+  workflow_call:
+    inputs:
+      git_ref:
+        type: string
+      location:
+        type: string
+        description: "the azure location to run the e2e test in"
+        default: "eastus"
+  #     k8s_version:
+  #       type: string
+  #       default: "1.27"
+    secrets:
+      E2E_CLIENT_ID:
+        required: true
+      E2E_TENANT_ID:
+        required: true
+      E2E_SUBSCRIPTION_ID:
+        required: true
+
+permissions:
+  contents: read
+        
+jobs:
+  initialize-generative-params:
+    runs-on: ubuntu-latest
+    outputs:
+      E2E_HASH: ${{ steps.generate-e2e-run-hash.outputs.E2E_HASH }}
+    steps:
+      - name: Harden Runner
+        uses: step-security/harden-runner@63c24ba6bd7ba022e95695ff85de572c04a18142 # v2.7.0
+        with:
+          egress-policy: audit
+
+      - id: generate-e2e-run-hash
+        run: |
+          E2E_HASH="$RANDOM$RANDOM"
+          echo "Using e2e hash \"$E2E_HASH\""
+          echo "E2E_HASH=$E2E_HASH" >> "$GITHUB_OUTPUT"
+  e2e:
+    needs: [initialize-generative-params]
+    strategy:
+      fail-fast: false
+      matrix:
+        suite: [Nonbehavioral, GPU, Integration, NodeClaim, Consolidation]
```

Comment:
This, and using the `e2e-nap` is really the only differences. I think its similar enough we could pass along a param on if its running in nap, or not

Reply:
At this point we could run one of two matrix sets based on nap or not. At least I think this is reasonable.